specs/models/StepSettings.v1.json

```diff
@@ -0,0 +1,157 @@
+{
+  "title": "StepSettings.v1",
+  "type": "object",
+  "description": "Settings for a step. This is a logical structure to simplify front-end/middle-tier APIs",
+  "properties": {
+    "customHook": {
+      "type": "object",
+      "properties": {
+        "module": {
+          "type": "string"
+        },
+        "parameters": {
+          "type": "string"
+        },
+        "user": {
+          "type": "string"
+        },
+        "runBefore": {
+          "type": "boolean"
+        }
+      }
+    },
+    "provenanceGranularityLevel": {
+      "type": "string",
+      "enum": [
+        "off",
+        "coarse",
+        "fine"
+      ]
+    },
+    "permissions": {
```

Comment:
I think since the UI is under development, we should use the same terms here. It's fine that we have to translate the new terms to the existing Flow schema, but for the UI, it'll never be easier than now to change it to use these new terms.